src/exoscale/vinyl/cursor.clj

```diff
@@ -32,18 +35,24 @@
                    (when (ifn? cont-fn)
                      (-> result .getContinuation .toBytes cont-fn))
                    (let [next?   (.hasNext result)
-                         new-acc (when next? (swap! acc f (.get result)))]
+                         new-acc (when next? (swap! acc reducer (.get result)))]
                      (and (not (reduced? new-acc)) next?))))))))
        (.getExecutor cursor))
-      (fn/make-fun (fn [_] (unreduced @acc))))))
+      (fn/make-fun (fn [_]
+                     (unreduced
+                      (cond-> @acc
+                        (some? xform)
+                        reducer)))))))
+  ([cursor f init cont-fn]
+   (apply-reduce cursor nil f init cont-fn))
```

Comment:
Shouldn't `apply-reduce` be renamed to `apply-transduce`, even if sometimes its just reducing?